crml/cennzx-spot/src/lib.rs

```diff
@@ -650,61 +665,69 @@ impl<T: Trait> Module<T> {
 
 		let asset_reserve = <generic_asset::Module<T>>::free_balance(asset_id, &exchange_address);
 		let core_reserve = <generic_asset::Module<T>>::free_balance(&core_asset_id, &exchange_address);
-		Ok(Self::get_input_price(
-			sell_amount,
-			asset_reserve,
-			core_reserve,
-			fee_rate,
-		))
+		Self::get_input_price(sell_amount, asset_reserve, core_reserve, fee_rate)
 	}
 
 	fn get_output_price(
 		output_amount: T::Balance,
 		input_reserve: T::Balance,
 		output_reserve: T::Balance,
 		fee_rate: FeeRate,
-	) -> T::Balance {
+	) -> rstd::result::Result<T::Balance, &'static str> {
 		if input_reserve.is_zero() || output_reserve.is_zero() {
-			return Zero::zero();
+			return Result::Err("Pool is empty");
 		}
 
 		// Special case, in theory price should progress towards infinity
 		if output_amount >= output_reserve {
-			return T::Balance::max_value();
+			return Ok(T::Balance::max_value());
 		}
 
-		let numerator: T::Balance = input_reserve * output_amount;
-		let denominator = output_reserve - output_amount;
-		let output: T::Balance = numerator / denominator + One::one();
+		let output_amount_u128: u128 = T::BalanceToU128::from(output_amount).into();
 
-		let output: T::AsBalance = output.into();
-		let output: u128 = output.as_();
-		let result = (FeeRate::one() + fee_rate) * output;
-		let result: T::AsBalance = As::sa(result);
-		Into::<T::Balance>::into(result)
+		let output_amount_uint = U256::from(output_amount_u128);
```

Comment:
Can take type hint out of all the variable names in the PR as it's redundant since there is support for type hinting in the language

Reply:
done